src/components/daos/daos.tsx

```diff
@@ -1,104 +1,186 @@
 "use client";
 
-import { useEffect, useState } from "react";
+import { useCallback, useEffect, useState, useRef } from "react";
+import { useQuery } from "@tanstack/react-query";
 import { Input } from "@/components/ui/input";
 import { supabase } from "@/utils/supabase/client";
-import Image from "next/image";
 import { Loader2, Search } from "lucide-react";
 import { Heading } from "@/components/ui/heading";
-import { Token } from "@/types/supabase";
-import { Card } from "@/components/ui/card";
-
-interface DAO {
-  id: string;
-  name: string;
-  mission: string;
-  description: string;
-  image_url: string;
-  is_graduated: boolean;
-  is_deployed: boolean;
-  created_at: string;
-  extensions?: Array<{
-    id: string;
-    type: string;
-  }>;
-}
+import { fetchTokenPrice } from "@/queries/daoQueries";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { DAOCard } from "./daos-card";
+import type { DAO, Token, SortField } from "@/types/supabase";
+import { createDaoAgent } from "../agents/dao-agent";
+import { useToast } from "@/hooks/use-toast";
+
+const fetchDAOs = async (): Promise<DAO[]> => {
+  const { data: daosData, error: daosError } = await supabase
```

Comment:
Noting this for the future as I really like how clear the flow here is (fetch, await, if error throw error), but if we need a performance bump we can batch the promises together with `Promise.all()` or a similar pattern.